src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -0,0 +1,91 @@
+import { SECONDS } from 'src/app/shared/helpers/duration';
+import { isNotUndefined } from 'src/app/shared/helpers/null-undefined-predicates';
+import { ItemTaskService } from '../../services/item-task.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { capitalize } from 'src/app/shared/helpers/case_conversion';
+import { ItemTaskInitService } from '../../services/item-task-init.service';
+import { ItemTaskAnswerService } from '../../services/item-task-answer.service';
+import { ItemTaskViewsService } from '../../services/item-task-views.service';
+
+const initialHeight = 1200;
+const additionalHeightToPreventInnerScrollIssues = 40;
+const heightSyncInterval = 0.2*SECONDS;
+
+interface TaskTab {
+  name: string,
+  view: string,
+}
+
+@Component({
+  selector: 'alg-item-display',
+  templateUrl: './item-display.component.html',
+  styleUrls: [ './item-display.component.scss' ],
+  providers: [ ItemTaskService, ItemTaskInitService, ItemTaskAnswerService, ItemTaskViewsService ],
+})
+export class ItemDisplayComponent implements OnInit, AfterViewChecked, OnChanges {
+  @Input() itemData?: ItemData;
+  @ViewChild('iframe') iframe?: ElementRef<HTMLIFrameElement>;
+
+  state$ = this.taskService.task$.pipe(mapToFetchState());
+
+  tabs$: Observable<TaskTab[]> = this.taskService.views$.pipe(
+    map(views => views.map(view => ({ view, name: this.getTabNameByView(view) }))),
+  );
+  activeTabView$ = this.taskService.activeView$;
+
+  iframeSrc$ = this.taskService.iframeSrc$;
+
+  // Start updating the iframe height to match the task's height
+  iframeHeight$ = merge(
+    this.taskService.task$.pipe(switchMap(task => interval(heightSyncInterval).pipe(switchMap(() => task.getHeight())))),
+    this.taskService.display$.pipe(map(({ height }) => height), filter(isNotUndefined)),
+  ).pipe(startWith(initialHeight), map(height => height + additionalHeightToPreventInnerScrollIssues));
+
+  constructor(
+    private taskService: ItemTaskService,
+  ) {}
+
+  ngOnInit(): void {
+    if (!this.itemData) throw new Error('itemData must be set in ItemDisplayComponent');
+    const url = this.itemData.item.url;
+    const attemptId = this.itemData.route.attemptId ?? this.itemData.currentResult?.attemptId;
+    if (!url || !attemptId) throw new Error('cannot load the task when item url or attempt is missing');
+
+    this.taskService.configure(this.itemData.route, url, attemptId);
+  }
+
+  ngAfterViewChecked(): void {
+    if (!this.iframe || this.taskService.initialized) return;
+    this.taskService.initTask(this.iframe.nativeElement);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (
+      changes.itemData &&
+      !changes.itemData.firstChange &&
+      (changes.itemData.previousValue as ItemData | undefined)?.item.id !== this.itemData?.item.id
+    ) {
+      throw new Error('This component does not support change of its itemData input');
+    }
+  }
+
```

Comment:
Now I re-read that, it seems wrong to me.  Actually itemData may change to the same item (same id) with additional info.  Maybe not right now, but in the branch I am working on, there is navigation info which may arrive later... also on "submit", the task might need reload the score/validated info... so it may change but for things which do not require a "reload" of item-display.
And BTW, sooner or later, the interface should allow to start/select a new attempt, so which will require a change... but that maybe handled with this change is introduced.

Maybe that's something related to https://github.com/France-ioi/AlgoreaFrontend/pull/474#discussion_r718223821, only relevant info for item-display should be given to item-display.

Reply:
Partly fixed by the other comment indeed, I added a check to throw only if the route item id changes
The route item id should not change because if it changes, the component is destroyed and re-created

That it handled by item-by-id which resets the itemData state to fetching when changing the route item id